library/bignum_mod.h

```diff
@@ -0,0 +1,142 @@
+/**
+ *  Internal bignum functions
+ *
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef MBEDTLS_BIGNUM_MOD_H
+#define MBEDTLS_BIGNUM_MOD_H
+
+#include "common.h"
+
+#if defined(MBEDTLS_BIGNUM_C)
+#include "mbedtls/bignum.h"
+#endif
+
+
+typedef struct
+{
+    size_t n;
+    mbedtls_mpi_uint *p;
+} mbedtls_mpi_mod_residue;
+
+typedef void* mbedtls_mpi_mont_struct;
+typedef void* mbedtls_mpi_opt_red_struct;
+
+typedef struct {
+    mbedtls_mpi_uint *p;
+    size_t n;       // number of limbs
+    size_t plen;    // bitlen of p
+    int ext_rep;    // signals external representation (eg. byte order)
```

Comment:
is "external representation" just "endianness", or can it be other things?

"e.g." suggests that this is just one thing it can for ("for example") but the enum only has LE and BE. And if it is just LE/BE, could we call this `endian` for clarity?

Reply:
No, it is not just endianness. LE keys and points are treated differently, also there is a special BE input when the source is hash and is read differently and we might want to be able to represent it.

For now we have only these two, but more are coming later.